src/io.c

```diff
@@ -2064,6 +2099,9 @@ _dispatch_disk_perform(void *ctxt)
 static void
 _dispatch_operation_advise(dispatch_operation_t op, size_t chunk_size)
 {
+#ifdef __LINUX_PORT_HDD__
```

Comment:
this should be a configure check for the `F_RDADVISE` fcntl (or short-term just an `#ifdef  F_RDADVISE`)